src/components/views/messages/MImageBody.js

```diff
@@ -472,12 +482,27 @@ export default class MImageBody extends React.Component {
           thumbUrl = this._getThumbUrl();
         }
 
-        const thumbnail = this._messageContent(contentUrl, thumbUrl, content);
-        const fileBody = this.getFileBody();
+        const collapsed = this.state.collapsed;
+        const thumbnail = collapsed ? null : this._messageContent(contentUrl, thumbUrl, content);
+        const fileBody = collapsed ? null : this.getFileBody();
+        /* when image is showing, it covers the collapse button, &nbsp; prevents it from overlaping */
+        const fileName = collapsed ? content.body : <div>&nbsp;</div>;
+
+        const chevronClasses = classNames({
+            'mx_RoomSubList_chevron': true,
+            'mx_RoomSubList_chevronRight': collapsed,
+            'mx_RoomSubList_chevronLeft': !collapsed,
```

Comment:
It would be better to split out the chevron to a common element if it's being used elsewhere, rather than using classes from a different component.

Reply:
Yes, you are correct.